auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -508,19 +520,32 @@ public MethodSpec createReadMethod(ClassName className,
       if (!prop.shouldDeserialize()) {
         continue;
       }
-      FieldSpec field = fields.get(prop);
-
-      for (String alternate : prop.serializedNameAlternate()) {
-        readMethod.addCode("case $S:\n", alternate);
+      if (prop.hasSerializedNameAnnotation()) {
+        FieldSpec field = fields.get(prop);
+        for (String alternate : prop.serializedNameAlternate()) {
+          readMethod.addCode("case $S:\n", alternate);
+        }
+        readMethod.beginControlFlow("case $S:", prop.serializedName());
+        readMethod.addStatement("$N = $N.read($N)", field, adapters.get(prop), jsonReader);
+        readMethod.addStatement("break");
+        readMethod.endControlFlow();
       }
-      readMethod.beginControlFlow("case $S:", prop.serializedName());
-      readMethod.addStatement("$N = $N.read($N)", field, adapters.get(prop), jsonReader);
-      readMethod.addStatement("break");
-      readMethod.endControlFlow();
     }
 
     // skip value if field is not serialized...
     readMethod.beginControlFlow("default:");
+    for (Property prop : properties) {
+      if (!prop.shouldDeserialize()) {
+        continue;
+      }
+      if (!prop.hasSerializedNameAnnotation()) {
+        FieldSpec field = fields.get(prop);
+        readMethod.beginControlFlow("if (renameField($S, gson).equals(_name))", prop.humanName);
```

Comment:
Since we're in the business of talking about caching, and the FieldNamingStrategy is immutable per TypeAdapter instance, instead of calling `renameField` on every serialize/deserialize pass, the renamed variant can be cached in a member variable on the TypeAdapter to speed up serialization.

Reply:
Would it be OK to do a single pass when the TypeAdapter is instantiated, store the renamed field names in a map, then reference those when seralizing/deserializing? I know there's been some previous work to lazy load elements within the TypeAdapter to improve performance so not sure if doing this when the class is created would also be considered too slow.

Reply:
That works for me.